pkg/prometheus/promcfg.go

```diff
@@ -1570,10 +1570,19 @@ func (cg *ConfigGenerator) generateProbeConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "scheme", Value: m.Spec.ProberSpec.Scheme})
 	}
 
+	var paramsMapSlice yaml.MapSlice
 	if m.Spec.Module != "" {
-		cfg = append(cfg, yaml.MapItem{Key: "params", Value: yaml.MapSlice{
-			{Key: "module", Value: []string{m.Spec.Module}},
-		}})
+		paramsMapSlice = append(paramsMapSlice, yaml.MapItem{Key: "module", Value: []string{m.Spec.Module}})
+	}
+
+	for _, p := range m.Spec.Params {
+		if p.Name != "module" {
+			paramsMapSlice = append(paramsMapSlice, yaml.MapItem{Key: p.Name, Value: p.Values})
+		}
```

Comment:
(suggestion) we can keep `module` if `.spec.module` is empty.
```suggestion
		if m.Spec.Module != "" && p.Name == "module" {
			continue
		}

		paramsMapSlice = append(paramsMapSlice, yaml.MapItem{Key: p.Name, Value: p.Values})
		}
```